### PR TITLE
Closes #235 - RS data for lymphoma

### LINKED ISSUE #235: General Issue: Add RS data for lymphoma
### Background Information

An RS dataset should be added for lymphoma response assessments.

It should include response assessments for PET metabolic response and CT response. For CT response standard response values (`"CR"`, `"PR"`, `"SD"`, ...) should be used for `RSSTRESC`. For PET response response values `"CMR"`, `"PMR"`, `"SMD"`, ... should be used.

The PET and CT assessment should be distinguished by the `RSSCAT` variable. Values: ?

For both assessments RESTESTCD should be ?

### Definition of Done

_No response_

### PR DESCRIPTION
**Pull request by kiran-peddamudium**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build pharmaversesdtm site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

To create the website for the content of this PR add "[create website]" to the title of the pull request.

### CODE REVIEW

File: R/rs_onco_lymphoma.R
Comment (bundfussr) [inline R/rs_onco_lymphoma.R:1]: Please move this file to `data-raw`.

File: R/rs_onco_lymphoma.R [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    cols = c("PET", "CT", "Overall"),
+    names_to = "assessment",
+    values_to = "RSORRES",
+    values_drop_na = FALSE
+  )
+
+testnames <- tibble::tribble(
+  ~assessment, ~RSTESTCD,  ~RSTEST,                               ~RSCAT,         ~RSSCAT,
+  "PET",       "PETRESP",  "Response (PET-CT)",                   "LUGANO 2014",  "INCLUDING PET-CT SCAN",
+  "CT",        "CTRESP",   "Response (CT or MRI without PET)",    "LUGANO 2014",  "NOT INCLUDING PET SCAN",
```
Comment (bundfussr) [inline R/rs_onco_lymphoma.R]: The `RSTESTCD` and `RSTEST` values are standardized by codelists C96782 and C96781 respectively. We should use values from the codelists.

We can discuss in the meeting which we want to use.

**Review by bundfussr — COMMENTED** [review]
Thanks @kiran-peddamudium  for setting up the new dataset!

 Here are some initial comments from my side:
 
- Please add an entry for the new dataset to `inst/extdata/sdtms-specs.json` and run `data-raw/create_sdtms_data.R` to generate the documentation.
- Please add an item to `NEWS.md`.
- You can also add your name to the authors in `DESCRIPTION`.

These points are not urgent.

File: data-raw/rs_onco_lymphoma.R [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  mutate(
+    RSDTC = sapply(strsplit(LBDTC, "T", fixed = TRUE), `[`, 1),
+    RSDY = LBDY
+  ) %>%
+  select(USUBJID, VISITNUM, RSDTC, RSDY)
+
+# Create rs_onco_lymphoma ----
+lymphoma_raw <- tibble::tribble(
+  ~USUBJID,      ~VISITNUM, ~VISIT,      ~PET,  ~CT,    ~OVRL, ~PETSTAT,
+  "01-701-1015", 1,         "BASELINE",  "NMR", "SAD",  "SD",  NA,
```
Comment (bundfussr) [inline data-raw/rs_onco_lymphoma.R]: Should we have baseline records in `RS`?

File: data-raw/rs_onco_lymphoma.R [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+# assign labels
+rs_onco <- pharmaversesdtm::rs_onco
+for (col in colnames(rs_onco_lymphoma)) {
+  if (col %in% colnames(rs_onco)) {
+    attr(rs_onco_lymphoma[[col]], "label") <- attr(rs_onco[[col]], "label")
+  }
+}
+
+rs_onco_lymphoma <- rs_onco_lymphoma %>%
+  add_labels(RSLOBXFL = "Last Observation Before Exposure Flag")
```
Comment (kiran-peddamudium) [inline data-raw/rs_onco_lymphoma.R]: @bundfussr Hi Stefan, I noticed that add_labels() is used in this, but the package for this function is not loaded, maybe we can use attr for consistency.

Comment (bundfussr) [reply data-raw/rs_onco_lymphoma.R]: As we decided to remove the baseline records, we can remove `RSLOBXFL` and this code.

File: data-raw/rs_onco_lymphoma.R (hunk 2 above)
Comment (kiran-peddamudium) [reply data-raw/rs_onco_lymphoma.R]: Removed baseline records

**Review by bundfussr — APPROVED** [review]
Thanks @kiran-peddamudium !
